### DISCUSSION TITLE
Raising the error (opposite of forwarding)

### DISCUSSION THREAD

**Discussion by micha149** [open-circle/valibot#795] [Q&A] [answered]
Hey,

I just migrated to the new Valibot API and wonder how to deal with the following situation. This is part of a form validation schema. The user enters a range of years (i.e., "2001-2006"). This range is parsed into a tuple of numbers, and I want to validate each number.

```ts
const referenceProjectSchema = v.pipe(
    v.object({
        id: v.pipe(
            v.unknown(),
            v.transform((s) => (s ? Number(s) : null)),
        ),
        description: v.pipe(v.string(), v.trim(), v.nonEmpty('Bitte gib eine Beschreibung ein.')),
        timeSpan: v.pipe(
            v.string(),
            v.transform(parseTimeSpan),
            v.tuple([
                v.pipe(
                    v.number(),
                    v.minValue(1950, 'Bitte gib sinnvolle Jahreszahlen ein.'),
                    v.maxValue(2100, 'Bitte gib sinnvolle Jahreszahlen ein.'),
                    v.check(
                        (input) => input <= new Date().getFullYear(),
                        'Der Zeitraum kann nicht in der Zukunft liegen.',
                    ),
                ),
                v.pipe(
                    v.number(),
                    v.minValue(1950, 'Bitte gib sinnvolle Jahreszahlen ein.'),
                    v.maxValue(2100, 'Bitte gib sinnvolle Jahreszahlen ein.'),
                ),
            ]),
            v.check(
                (input) => input[0] <= input[1],
                'Bitte gib ein oder zwei Jahreszahlen in der richtigen Reihenfolge ein.',
            ),
        ),
    }),
    v.transform(({ timeSpan, ...rest }) => ({ ...rest, startYear: timeSpan[0], endYear: timeSpan[1] })),
);
```

If any validation fails, I get a path like `timeSpan.0`, but my form only knows `timeSpan`, so it would be nice to be able to raise the error to the `timeSpan` property. On the other hand, I wonder if paths shouldn't refer to the original data instead of the transformation result of any action.

Or is there something obvious I'm missing?

**Comment by fabian-hiller** [comment, marked as answer]
Your goal requires that you validate the numbers in the outer pipeline. Here is an example. Check it out in [this playground](https://valibot.dev/playground/?code=JYWwDg9gTgLgBAKjgQwM5wG5wGZQiOAcg2QBtgAjCGQgbgCh6BjCAO1XgBVQBTAZTDJWfJgAseIZHAC8mAHRhgYHgAp6ceRyjBWAcxUBKADTr5UHrp4APNRo0B6AHoAdACYBvACwBfALRuvbwASewBXEzsiACFgGBgeOF1KOAAvAHceYDgAKWRRc1QUvNIeVjhQOAAxaEl4ACIATSaG32aGurhM1jlCCLhjUww5GCghVGwalRUdMFCYAxkAPnLWWZg5VDByGBVCX0IDOUkwFQA5UJAKHigFtDgAbVYLq6gjOCfL64BdAY0hngw1wAngBJeIgWx2aarOYLaTLGZzOCLWQARgAnABWAAMcAAZHiVms4AAeWQAJlR2OxfQ0hBicQSSQocFQOlYGAgpBKOTyBSKohKZS6PT6v3kYh4TAA1pCNNC1nCETCYPdsV9SbJEarUV9adFYvFEslXMAErl8jxCsVSis4K5rnBtGIYMBLGUAEqZcSsCakSydHSi0wDAwMZhsDhOq2hUjwWRDVDIbA8AAKyCgqFU3BA-EEwklkjehHJ1OxvlL2IAbAdwyx2FyeHJSBB9AVY-MGEA).

```ts
import * as v from 'valibot';

const TimeSpanSchema = v.pipe(
  v.string(),
  v.regex(
    /^\d{4}-\d{4}$/u,
    'Bitte gib zwei Jahreszahlen im Format "YYYY-YYYY" ein.',
  ),
  v.transform((input) => input.split('-').map(Number) as [number, number]),
  v.everyItem(
    (input) => input >= 1950 && input <= 2100,
    'Bitte gib sinnvolle Jahreszahlen ein.',
  ),
  v.check(
    (input) => input[0] <= input[1],
    'Bitte gib die Jahreszahlen in der richtigen Reihenfolge ein.',
  ),
);
```

**Reply by micha149** [reply]
Thanks a lot for your prompt help! I was initially aiming for a more declarative approach, as I find it more readable. However, your suggestion works well for me, so I'll proceed with it. I've just added a variant that handles a single year as well.

Thanks again for your support!
